src/main/java/org/apache/sysml/runtime/codegen/LibSpoofPrimitives.java

```diff
@@ -1379,7 +1379,72 @@ public static void vectMult2Add(double[] a, double[] c, int[] aix, int ai, int c
 			c[aix[j]] = a[j] + a[j];
 		return c;
 	}
-	
+
+	//custom vector xor
+    //TODO sparse operations are not working correctly
+	public static void vectXorAdd(double[] a, double bval, double[] c, int ai, int ci, int len) {
+		for( int j = ai; j < ai+len; j++, ci++)
+			c[ci] +=  ( (a[j] != 0) != (bval != 0) ) ? 1 : 0;
+	}
+
+	public static void vectXorAdd(double bval, double[] a, double[] c, int ai, int ci, int len) {
+		for( int j = ai; j < ai+len; j++, ci++)
+			c[ci] += ( (bval != 0) != (a[j] != 0) ) ? 1 : 0;
+	}
+
+	public static void vectXorAdd(double[] a, double bval, double[] c, int[] aix, int ai, int ci, int alen, int len) {
+		for( int j = ai; j < ai+alen; j++ )
+			c[ci + aix[j]] += ( a[j] != bval ) ? 1 : 0;
+	}
```

Comment:
This computation is incorrect: for simplicity, I would recommend to first add xor(bval,0) from all entries of c and subsequently make a pass over the sparse a and subtract 1 if (a[j]!=0) == (bval!=0). Note that the current computation `( a[j] != bval ) ? 1 : 0` is NOT xor.